includes/collection/class-outbox.php

```diff
@@ -74,4 +74,23 @@ public static function add( $activity_object, $activity_type, $user_id, $content
 
 		return $id;
 	}
+
+	/**
+	 * Get pending activities.
+	 *
+	 * @param int $limit The number of activities to get.
+	 *
+	 * @return WP_Post[] The pending activities.
+	 */
+	public static function get_pending( $limit = 10 ) {
+		$activites = \get_posts(
+			array(
+				'post_type'      => self::POST_TYPE,
+				'post_status'    => 'pending',
+				'posts_per_page' => $limit,
```

Comment:
```suggestion
				'posts_per_page' => $limit,
				'fields'         => 'ids',
```
Since we only need IDs in `reprocess_outbox`, we can make this query faster by not asking for the whole object.